### PR TITLE
implement a changes by repository box in the ui

### PR DESCRIPTION
**Pull request by flepied**

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -91,6 +92,12 @@ class RootView extends React.Component {
                   <CLastChanges />
                 </Col>
               </Row>
+              <Row>
+                <Col>
+                  <CRepoChanges />
+                </Col>
+              </Row>
+              <Row><Col><p></p></Col></Row>
```
Comment (morucci) [inline web/src/App.js:100]: can be removed

File: web/src/App.js [hunk 2]
```diff
@@ -91,6 +92,12 @@ class RootView extends React.Component {
                   <CLastChanges />
                 </Col>
               </Row>
+              <Row>
+                <Col>
+                  <CRepoChanges />
```
Comment (morucci) [inline web/src/App.js:97]: I'm wondering to have a repository related events tab. Now we have Events stats, changes status, what about a Repo stats. Even if we will rework layout and where graph/stat should be, I think there is a room for that new tab:

- Top repository by merged changes
- Top repository by open changes
- Top 3 authors by repo
- Less active repositories
- ...

File: web/src/components/common.js [hunk 3]
```diff
@@ -12,6 +12,24 @@ function change_url(x, name=null) {
   return <a href={x.url} target="_blank" rel="noopener noreferrer">{name}</a>;
 }
 
+function filter_url(url, field) {
```
Comment (morucci) [inline web/src/components/common.js:15]: You should use this: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get

File: web/src/components/common.js [hunk 4]
```diff
@@ -12,6 +12,24 @@ function change_url(x, name=null) {
...
+  
+  if (url_parts.length > 1) {
+    const new_args = url_parts[1].split('&').filter(x => !x.startsWith(field + '='));
+    
+    return url_parts[0] + '?' + new_args.join('&');
+  }
+  return url;
+}
+
+function add_url_field(url, field) {
```
Comment (morucci) [inline web/src/components/common.js:26]: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set

Comment (morucci) [reply web/src/components/common.js:26]: See also here, its already used: https://github.com/morucci/monocle/blob/master/web/src/components/filtersform.js#L77

File: monocle/db/queries.py [hunk 5]
```diff
@@ -249,6 +249,12 @@ def _events_top(
         'count_avg': count_avg, 'count_median': count_median}
 
 
+def merged_changes_by_repository(es, index, repository_fullname, params):
```
[the remark this answers is not in the document]
Comment (flepied) [reply monocle/db/queries.py:252]: ok done

File: web/src/App.js (hunk 2 above)
Comment (flepied) [reply web/src/App.js:97]: yes good idea but in following PR

File: web/src/App.js (hunk 1 above)
Comment (flepied) [reply web/src/App.js:100]: done

File: web/src/App.js (hunk 2 above)
Comment (flepied) [reply web/src/App.js:97]: yes in a following PR

File: web/src/components/common.js (hunk 3 above)
Comment (flepied) [reply web/src/components/common.js:15]: ok done

File: web/src/components/common.js (hunk 4 above)
Comment (flepied) [reply web/src/components/common.js:26]: done

**Pull request merged**